src/validation.cpp

```diff
@@ -3641,17 +3640,20 @@ CBlockIndex * CChainState::InsertBlockIndex(const uint256& hash)
     if (hash.IsNull())
         return nullptr;
 
+    LOCK(cs_main);
```

Comment:
This should absolutely be an AssertLockHeld, if anything. If you add the LOCK(cs_main) in the block-index-loading code in init.cpp this should be able to go away.

Reply:
Do we want to lock that long?

Reply:
It doesnt matter, its early in init long before anything else is going to take cs_main, so taking it should have no impact.